### ISSUE TITLE
MicrosoftTeams.munki.recipe problem with updating

### ISSUE THREAD

**Issue by opus-nbo** [autopkg/apizz-recipes#66]
Hi,

Our fleets clients are installing Teams over and over again because not all receipts are there (audio driver not installed). 

Either, this recipe needs the `com.microsoft.MSTeamsAudioDevice` receipt set to be optional or there must be an installer choice XML that installs both, `com.microsoft.MSTeamsAudioDevice` and `com.microsoft.teams`.

I would say it makes sense to do the latter, as Teams would be more complete with audio drivers installed.
Can you make this adjustments, @apizz? Thanks!

**Comment by apizz** [comment]
Hi, @opus-nbo.  Admittedly I don't use teams anymore, but in this case for an easy unblock you could just put an `installer_choices_xml` array in your recipe override's Input variables without needing any other changes to the recipe processing.  While this would be helpful for other admins to have in the recipe template by default, it shouldn't be a blocker for you or others for addressing this.

ex input vars:
```
<key>Input</key>
<dict>
    <key>MUNKI_CATEGORY</key>
    <string>Communication</string>
    <key>MUNKI_REPO_SUBDIR</key>
    <string>apps/microsofteams</string>
    <key>NAME</key>
    <string>MicrosoftTeams</string>
    <key>pkginfo</key>
    <dict>
        <key>catalogs</key>
        <array>
            <string>testing</string>
        </array>
        <key>category</key>
        <string>%MUNKI_CATEGORY%</string>
        <key>description</key>
        <string>Microsoft Teams</string>
        <key>developer</key>
        <string>Microsoft</string>
        <key>display_name</key>
        <string>Microsoft Teams</string>
        <key>name</key>
        <string>%NAME%</string>
        <key>unattended_install</key>
        <true/>
        <key>installer_choices_xml</key>
        <array>
            <dict>
                <key>attributeSetting</key>
                <integer>1</integer>
                <key>choiceAttribute</key>
                <string>selected</string>
                <key>choiceIdentifier</key>
                <string>com.microsoft.teams</string>
            </dict>
            <dict>
                <key>attributeSetting</key>
                <integer>1</integer>
                <key>choiceAttribute</key>
                <string>selected</string>
                <key>choiceIdentifier</key>
                <string>com.microsoft.MSTeamsAudioDevice</string>
            </dict>
        </array>
    </dict>
</dict>
```

That said, reviewing my recipes there's improvements to be made specifically for the munki side, since we can base the install on the presence of the MS teams app, rather than the pkg receipts (which I've personally always preferred).

If you pull my latest recipe repo and create a new override you should be able to ensure both PKGs get installed _and_ base the Teams install on the existence of the app, rather than the pkg receipts.

**Comment by opus-nbo** [comment]
@apizz I noticed your changes to the recipe. Are you sure the identifiers in the `installer_choice_xml` are correct?

I did a `installer -showChoicesXML` on the Teams pkg and the resulting identifiers are:

```
						<key>choiceIdentifier</key>
						<string>TeamsApp</string>
```

```
				<key>choiceIdentifier</key>
				<string>Teams</string>
```

```
				<key>choiceIdentifier</key>
				<string>AudioDevice</string>
```

**Comment by apizz** [comment]
Like I said I don't use teams anymore but as indicated those choice options are input variables which are entirely admin configurable.  I reviewed the installer and got the receipts from Suspicious Package, so they should reflect reality.

If these are inaccurate please submit a PR for what they should be and provide output or screenshots that shows what they should be.

I won't be making any more changes here without additional guidance since I don't use teams anymore.

**Comment by apizz** [comment]
At a core level though, the problem you originally indicated regarding updating should be fixed.

**Comment by opus-nbo** [comment]
@apizz I already checked myself with my own recipe, first i used the same identifiers as you, then i checked the choicesXML and found out that they in fact do not correspond with the package identifiers. After testing the new identifiers, the audio driver was installed successfully. I think you can safely make the change.

**Comment by apizz** [comment]
Then this is very easy. I will not be making the change however until you have shown the actual changes to be made and successful recipe run output and not within this issue, which while related has been resolved.

1. Please file a new PR with your recipe changes.
2. Please include a verbose recipe run that shows this works as expected.

This is how you contribute to open source code.
